doc/dynamo0p3.rst

```diff
@@ -1256,6 +1336,27 @@ and at W1)::
        local_stencil, xdata, ydata, zdata, ndf_w0, undf_w0, map_w0, &
        basis_w0_on_w0, basis_w0_on_w1, ndf_w1)
 
+If the metadata specifies that the kernel requires a property of the
+reference element::
+
+  type, extends(kernel_type) :: testkern_operator_type
+     type(arg_type), dimension(2) :: meta_args =      &
+          (/ arg_type(gh_operator, gh_write, w0, w1), &
+             arg_type(gh_field*3, gh_read, w0) /)
+     type(reference_element_data_type) :: meta_reference_element(1) =  &
```

Comment:
Since we need to explicitly pass the number of faces as well, I would suggest
```
     type(reference_element_data_type) :: meta_reference_element(2) =  &
          (/ reference_element_data_type(number_horizontal_faces),     &
             reference_element_data_type(normal_to_face) /)
```

Reply:
We do but that's implicit in the fact that the user is requesting the face normals - we shouldn't require them to supply effectively duplicate information.

Reply:
We are not supplying duplicate information here because the faces are not necessarily horizontal (see above comments) so the number of faces can be horizontal, vertical or all.
If we passed only `normal_to_horizontal_face` (which we do not have in the LFRic infrastructure) then `number_horizontal_faces` is implicit and we do not need to pass it.